src/IncentiveLevelSlipContainer/IncentiveLevelSlipContainer.js

```diff
@@ -12,7 +15,32 @@ function IncentiveLevelSlipContainer() {
   const [printed, setPrinted] = useState(false)
 
   useEffect(() => {
-    setIepData(JSON.parse(localStorage.getItem('incentiveLevelSlip')))
+    const setUpData = async () => {
+      const data = localStorage.getItem('incentiveLevelSlip')
+      if (data) {
+        setIepData(JSON.parse(data))
+      } else if (!printed && !iepData) {
```

Comment:
Without these checks, the component kept checking, and kept making the request to the api and all of the below. This makes sure it only does the thing when it needs to.